### PR TITLE
UI2 create account set up pin

### PR DESCRIPTION
**Pull request by dbo1dyrev**
PR Requirements
If you have made any visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS Tablet
- [x] Tested on small Android
- [ ] n/a
Screenshot 2021-06-16 at 18 52 34
Screenshot 2021-06-16 at 18 52 44
Screenshot 2021-06-16 at 18 52 54

### CODE REVIEW

**Comment by paullinator** [comment]
Remove the red warning box that the pin must be four digits. This was supposed to be removed from the design.

**Review by paullinator — CHANGES_REQUESTED** [review]
See comment. This is not a thorough review but I at least want that UI change

**Comment by dbo1dyrev** [comment]
Screenshot 2021-06-17 at 17 15 13

>Remove the red warning box that the pin must be four digits. This was supposed to be removed from the design.

fixed it

File: packages/edge-login-ui-rn/src/components/themed/FourDigitInput.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  return (
+    <TouchableWithoutFeedback onPress={handleRefocus}>
+      <View style={[styles.container, spacings]}>
+        <View style={styles.interactiveContainer}>
+          {wait > 0 ? (
+            <ActivityIndicator color={theme.iconTappable} size="large" />
+          ) : (
+            <View style={styles.dotContainer}>
```
Comment (thehobbit85) [inline packages/edge-login-ui-rn/src/components/themed/FourDigitInput.js]: You should try and avoid writing code like this.
When you have a repeating pattern, you should generalize it like here: 

https://github.com/EdgeApp/edge-react-gui/blob/develop/src/components/themed/PinDots.js

And in this case you should also separate the "PinDots" component like the one in the link.

File: packages/edge-login-ui-rn/src/components/themed/FourDigitInput.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+// @flow
```
Comment (thehobbit85) [inline packages/edge-login-ui-rn/src/components/themed/FourDigitInput.js:1]: The word "Four" shouldn't be a part of any file/folder/component name.
We might change it tomorrow to 5, or 6, or whatever. It should be kept general and the length of the Pin Code should be a prop.

File: packages/edge-login-ui-rn/src/components/screens/newAccount/NewAccountPinScreen.js [hunk 3]
```diff
@@ -1,180 +1,158 @@
...
+        >{`${s.strings.step_three}: ${s.strings.set_four_digit_pin}`}</EdgeText>
+        <EdgeText style={styles.description} numberOfLines={2}>
+          {s.strings.pin_desc}
+        </EdgeText>
+        <FourDigitInput />
+        <FormError>{pinErrorMessage}</FormError>
+        <View style={styles.actions}>
+          <Fade
+            visible={
+              pin.length === 4 && !pinErrorMessage && !createErrorMessage
```
Comment (thehobbit85) [inline packages/edge-login-ui-rn/src/components/screens/newAccount/NewAccountPinScreen.js:100]: Avoid using numbers directly in the code.
Always pass them as a Param/Constant.

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comments

**Comment by dbo1dyrev** [comment]
add new commit "Add Pin Dots component"
rebase "Change the PIN input design to match the new design"

**Comment by dbo1dyrev** [comment]
Screenshot 2021-07-01 at 13 02 17

File: packages/edge-login-ui-rn/src/components/screens/newAccount/NewAccountPinScreen.js [hunk 4]
```diff
@@ -1,180 +1,158 @@
...
+    setIsProcessing(true)
+
     // validation.
-    // is there no error message ,
-    if (this.props.pin.length !== 4 || this.props.pinError) {
-      this.setState({
-        isProcessing: false
-      })
+    // is there no error message,
+    if (pin.length !== 4 || pinErrorMessage) {
```
Comment (thehobbit85) [inline packages/edge-login-ui-rn/src/components/screens/newAccount/NewAccountPinScreen.js:72]: Another magic number

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
Magic number again

**Comment by dbo1dyrev** [comment]
> /rebase-autosquash

already did it

**Pull request merged**